### PR TITLE
Add Zolertia Firefly board

### PR DESCRIPTION
**Pull request by vandaele**
Adds support for the (Zolertia Firefly)[https://github.com/Zolertia/Resources/wiki/Firefly] board.

This boards requires the cc2538 Boot Serial Loader to flash firmware. Support in the gateway code and install instructions have been added, using https://github.com/JelmerT/cc2538-bsl.

### CODE REVIEW

File: gateway_code/utils/CC2538.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ class CC2538(object):
     DEBUG = ('')
 
     @property
-    def add_command(self):
+    def addr_command(self):
+        #TODO Do it a better way with python
```
Comment (vandaele) [inline gateway_code/utils/CC2538.py]: There's a TODO here.
Could be better to replace this long and complex shell command above by python code.

File: bin/rules.d/firefly.rules [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# open nodes firefly
+# previous work by Cedric
+# KERNEL=="ttyUSB[0-9]*" , SUBSYSTEMS=="usb", ATTRS{idVendor}=="10c4" , ATTRS{idProduct}=="ea60" , ATTRS{product}=="Zolertia Firefly platform" , MODE="0664", GROUP="dialout", SYMLINK+="ttyON_firefly"
```
Comment (rienafairefr) [inline bin/rules.d/firefly.rules]: This was taken from http://zolertia.sourceforge.net/wiki/index.php/Mainpage:Apps

File: setup.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def get_version(package):
 SCRIPTS = glob('bin/scripts/*')
 
 INSTALL_REQUIRES = ['argparse', 'bottle', 'paste', 'pyserial']
-INSTALL_REQUIRES += ['pyelftools']
+INSTALL_REQUIRES += ['pyelftools', 'intelhex', 'python-magic']
```
Comment (aabadie) [inline setup.py]: Those new packages don't seem to be used in the changes. Are they really required ?

Comment (rienafairefr) [reply setup.py]: it's due to the cc2538 dependency. 
python-magic is not fully needed, I think I'll just write temporary firmwares to flash with a .hex extension and not add python-magic.

Comment (aabadie) [reply setup.py]: you mean the cc2538-bsl.py tool ? this is an external tool so no need to add this dependency to the gateway code. I think it's enough to just have it as a dependency in the docker container and a note in the readme (like it's already done)

Comment (rienafairefr) [reply setup.py]: You're right

**Comment by rienafairefr** [comment]
OK this has been tested in local, code_check. Normally the latest `master` yocto images already have the cc2538-bsl added so we could merge that and plug a node in ci

**Comment by aabadie** [comment]
Let's go then !

**Pull request merged**